HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+# HIP-0006 : Trustless Live Auctions for Handshake
+
+```
+Number:  HIP-0006
+Title:   Trustless Live Auctions for Handshake
+Type:    Informational
+Status:  Final
+Authors: Matthew Zipkin (@pinheadmz)
+Created: 2021-06-17
+```
+
+## Abstract
+
+We propose an ascending-price auction system for HNS domain names without trust or
+third-party custody. A name owner initiates an auction by locking their TLD with
+a verifiable script. Bidders place bids by partially-signing `TRANSFER` transactions,
+and the name owner eventually picks the highest bid and atomically swaps the name
+ownership for the payment. The protocol requires more interactivity and centralization
+than the current Shakedex-style HIP-0001 auction format, but with some compelling trade-offs.
+
+## Definitions
+
+This proposal is based on [HIP-0001](https://hsd-dev.org/HIPs/proposals/0001/), and
+in particular the Swap Script, Transaction Structure, and Swap Proof. Handshake
+primitives like `OP_TYPE` and `SIGHASH_SINGLEREVERSE` are also required.
+
+Recall the HIP-0001 Swap Script only allows two covenant actions be applied to the name:
+
+- The name owner can sign a TRANSFER with their private key
+
+- Anyone can FINALIZE without a signature
+
+## Protocol Summary
+
+### Auction OPEN
+
+The auction technically begins when a name owner FINALIZEs a transfer of their name
+to a HIP-0001 locking script. The name owner must then publicly reveal the redeem script
+for the address that owns the name, so it can be verified. The locking script prevents
+the name owner from canceling an in-progress name ownership transfer. Since there is a
+small amount of interactivity required in the protocol, the name owner must also publish
+"where" the auction will take place. The auction "location" is just some internet endpoint
```

Comment:
"must also publish" sounds a bit too strong, could be misunderstood to be a requirement? Why not relax it to something like "it is up to the name owner to communicate the existence of this auction and to provide means for potential buyers to submit bids using mechanisms that are outside of the scope of this proposal. For example, ...."

Reply:
Hm, I understand your sentiment but it is actually a requirement of this protocol.